### PR TITLE
(Closes #2622) Add convenient methods `SymbolTable.insert_argument` and `.append_argument`.

### LINKED ISSUE #2622: Manipulating/updating Routine arguments
It is currently difficult to manipulate the arguments of a `Routine` because of the internal consistency checks that are performed by the setter. This needs revisiting.

### PR DESCRIPTION
**Pull request by JulienRemy**
Arguments currently have to be inserted in `SymbolTable().argument_list` using a workaround: 
- 1. getting it, which checks the arguments and non-arguments symbols in the table,
- 2.  adding the symbol to the table and the argument list. Trying to do it in the reversed order causes a validation error on either the table or the argument list.

This introduces `SymbolTable().insert_argument(index, argument)` and `.append_argument(argument)` to hopefully cover most use cases.

### CODE REVIEW

**Comment by JulienRemy** [comment]
@arporter @sergisiso What do you think about these?

**Review by arporter — CHANGES_REQUESTED** [review]
Thanks very much for this Julien. Just small things to tidy up. 
Updated test file fully covers modified source file.
No need to update user guide or dev guide because we don't mention the management of Routine argument lists! While that's not good, I don't think we have to address that here.
Ref. guide builds without extra warnings.
Although this PR really shouldn't affect anything, I'll launch the integration tests just to be on the safe side.

File: src/psyclone/tests/psyir/symbols/symbol_table_test.py [hunk 1]
```diff
@@ -1582,6 +1582,136 @@ def test_argument_list_errors():
...
+                in str(err.value))
+
+    sym_table = symbols.SymbolTable()
+    var1 = symbols.DataSymbol("var1", symbols.REAL_TYPE)
+    sym_table._argument_list.append(var1)
+    arg1 = symbols.DataSymbol("var2", symbols.REAL_TYPE,
+                              interface=symbols.ArgumentInterface())
+    with pytest.raises(InternalError) as err:
+        sym_table.insert_argument(0, arg1)
+        assert ("var1" in str(err.value)
```
[the remark this answers is not in the document]
Comment (JulienRemy) [reply src/psyclone/tests/psyir/symbols/symbol_table_test.py:1646]: Do you mean line 1643?
Unindenting to the previous tab i.e.
```python
    arg1 = symbols.DataSymbol("var2", symbols.REAL_TYPE,
                            interface=symbols.ArgumentInterface())
    with pytest.raises(InternalError) as err:
        sym_table.insert_argument(0, arg1)
        assert ("var1" in str(err.value)
```
causes Flake8 to complain: `src/psyclone/tests/psyir/symbols/symbol_table_test.py:1643:29: E128 continuation line under-indented for visual indent`

[the remark this answers is not in the document]
Comment (JulienRemy) [reply src/psyclone/tests/psyir/symbols/symbol_table_test.py:1646]: Same with even one space less.

File: src/psyclone/tests/psyir/symbols/symbol_table_test.py [hunk 2]
```diff
@@ -1582,6 +1582,136 @@ def test_argument_list_errors():
...
+            str(err.value))
+
+    arg3 = symbols.DataSymbol("var6", symbols.REAL_TYPE,
+                              interface=symbols.ArgumentInterface())
+    sym_table.add(arg3)
+    arg4 = symbols.DataSymbol("var7", symbols.REAL_TYPE,
+                              interface=symbols.ArgumentInterface())
+    with pytest.raises(InternalError) as err:
+        sym_table.append_argument(arg4)
+        assert ("var6" in str(err.value)
```
[the remark this answers is not in the document]
Comment (JulienRemy) [reply src/psyclone/tests/psyir/symbols/symbol_table_test.py:1696]: Same as above.

**Comment by JulienRemy** [comment]
`assert`s are now unindented and I also changed the `_validate_...` error strings to use `type(object).__name__`.

**Pull request merged**